### ISSUE TITLE
bug in rustls-mio

### ISSUE THREAD

**Issue by bradyjoestar** [ctz/rustls#285]
I have found the following situation:
```
number of connections is: 19
self closed: false
self closing: true
self tls_session.wants_write: false
reregister
number of connections is: 19
self closed: false
self closing: true
self tls_session.wants_write: false
reregister
number of connections is: 19
self closed: false
self closing: true
self tls_session.wants_write: false
reregister
number of connections is: 19
self closed: false
self closing: true
self tls_session.wants_write: false
reregister
number of connections is: 19
self closed: false
self closing: true
self tls_session.wants_write: false
reregister

```

The client is closed and the sever is trapped in loop of `conn_event`  and `ready` function.

https://github.com/ctz/rustls/blob/056d843c0b96ea5ccf7837da3b2de33a87fc0c28/rustls-mio/examples/tlsserver.rs#L182-L189

Maybe we should change the above code to 
```
if self.closing {
            let _ = self.socket.shutdown(Shutdown::Both);
            self.close_back();
            self.closed = true;
        } else {
            self.reregister(poll);
        }
```

Thanks!

**Comment by ctz** [comment]
Thanks for the report; fixed in a93ee1ab

**Comment by ctz** [comment]
This has been assigned CVE-2019-15541, I will inform rustsec.
